src/toil_scripts/lib/programs.py

```diff
@@ -123,11 +123,12 @@ def docker_call(tool,
                 return subprocess.check_output(docker_call)
             else:
                 subprocess.check_call(docker_call)
-
-    except subprocess.CalledProcessError:
-        raise RuntimeError('docker command returned a non-zero exit status. Check error logs.')
-    except OSError:
-        raise RuntimeError('docker not found on system. Install on all nodes.')
+    finally:
```

Comment:
Sorry for not catching this earlier: The problem with `finally:` is that an exception raised there will mask any exception that occurred in `try:`

```
>>> try:
...     raise ValueError()
... finally:
...     raise RuntimeError()
...     
Traceback (most recent call last):
  File "<input>", line 4, in <module>
RuntimeError
```

There are various ways around this, none of them pretty, caused by the fact that https://docs.python.org/2/library/sys.html#sys.exc_info() is backed by a single thread-local variable.

```
>>> from bd2k.util.exceptions import panic
>>> try:
...     raise ValueError()
... except:
...     with panic():
...         raise RuntimeError()
... else:
...     pass
...     
Traceback (most recent call last):
  File "<input>", line 5, in <module>
  File "<input>", line 2, in <module>
ValueError
```

The easiest would be to omit the `try: finally:` altogether and accept the fact that on exceptions, permissions might be wrong. But I have the feeling that that won't work for us. bd2k-python-lib's panic() context manager is the next best thing I can think of. It would require that you duplicate the permission fix in both the `except:` (enclosed in `panic()`) and the `else:` (without `panic()`). That can be alleviated by extracting that code into a helper function.